main/main.cpp

```diff
@@ -725,11 +725,16 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 			forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->get());
 			forwardable_cli_arguments[CLI_SCOPE_PROJECT].push_back(I->get());
 		}
-		if (I->get() == "--single-window" ||
-				I->get() == "--audio-driver" ||
+		if (I->get() == "--single-window") {
+			forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->get());
+		}
+		if (I->get() == "--audio-driver" ||
 				I->get() == "--display-driver" ||
 				I->get() == "--rendering-driver") {
-			forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->get());
+			if (I->next()) {
+				forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->get());
+				forwardable_cli_arguments[CLI_SCOPE_TOOL].push_back(I->next()->get());
```

Comment:
Does the `TOOL` scope include running the project from the editor too? As if forcing a rendering driver that overrides the one from project settings for the editor, it should likely also be used when running.

Reply:
It's only for the project manager and editor restarts. I added the concept of scopes precisely to avoid such arguments from crossing the editor-project boundary. My only concern is the potential of confusion when a user switches drivers in project settings, but they don't get applied because of the original CLI argunments being "stuck". That said, I don't really have a strong opinion on it. Just did what I considered sensible.

In any case, the PR where this was introduced already had a decision on this and this one is only meant to fix it. If we finally decide it's better that all the arguments make it to the running game, that'd be matter for a subsequent PR.

Reply:
Am tending to be in agreement with @RandomShaper , although there are some cases for passing arguments to the running project in the editor, it is a bit of a bag of worms and might be better in a separate PR (if we decided to do it).

For running projects from the editor, maybe we could have a command line section in the project settings, or some checkboxes / toggles for the editor runs? :thinking: